### ISSUE TITLE
`ag -g <filename>` does not use smart case

### ISSUE THREAD

**Issue by ptzz** [ggreer/the_silver_searcher#891]
`ag -S -g <lower_case_query>` matches when `ag -g <lower_case_query>` does not. According to the help they should be the same as `-S` is enabled by default:

> -S --smart-case Match case insensitively unless PATTERN contains uppercase characters (Enabled by default)

**Comment by jnboehm** [comment]
I couldn't reproduce this with
```
touch Bla
touch bla
ag -g bla
```
returns both file names for me

**Comment by ptzz** [comment]
I'm on OSX (10.11.4) and cannot create both Bla and bla. 
```
test~/tmp$ ag --version
ag version 0.31.0

Features:
  +jit +lzma +zlib
test~/tmp$ touch Bla && ag -g bla
test~/tmp$ rm Bla
test~/tmp$ touch bla && ag -g bla
bla
```

**Comment by ptzz** [comment]
It works on latest master, `brew install --HEAD ag` fixed it

**Comment by ptzz** [comment]
Fixed by #898

**Comment by Fuco1** [comment]
I don't think this works correctly still.  When I use just `ag -g 'SmartPattern'` it does what I would expect, when I attach a path so that it becomes `ag -g 'SmartPattern' /search/here` it works the same.

But if I use `-S` with a path then it reverts back to case-insensitive search.
